### PR TITLE
Mapping - basic changes

### PR DESCRIPTION
**Pull request by kaijli**
@aclum @scanon successful run of mapping task, jaws id 79113
the other mapping branch got too convoluted, so this is from an older commit on that branch.

### CODE REVIEW

**Comment by kaijli** [comment]
https://github.com/microbiomedata/mg_annotation/pull/27#pullrequestreview-2023122092
In response to this previous change request:

> 1.    For this to follow nmdc naming conventions we need to two identifier variables, one for the assembly identifier (ex nmdc:wfmgas-11-0080kf20.1) and one for the annotation identifier (ex nmdc:wfmgan-11-001dky13.1). The assembly identifier is what should get passed in to the make_map_file function for the generation of the mapping file. The annotation file is what should get passed to finish_ano which renames the annotation file names.

@scanon @aclum not sure if this was ever discussed fully or ironed out, but can be an easy thing to implement. I think Shane was looking at how / whether to pull the assy-id from somewhere during automation

> 2.    The newly named nucleotide fasta file and the mapping file need to be output files for finish_ano. The newly named contig file that is an output of make_map_file should have a suffix of _contigs.fna and the mapping file should have a suffix of _contig_names_mapping.tsv

Done

**Review by aclum — APPROVED** [review]

File: annotation_full.wdl [hunk 1]
```diff
@@ -222,10 +240,11 @@ task stage {
    String input_file
 
    command <<<
-       set -e
+       set -euo pipefail
        if [ $( echo ${input_file}|egrep -c "https*:") -gt 0 ] ; then
            wget ${input_file} -O ${target}
        else
+           set +o pipefail
```
Comment (aclum) [inline annotation_full.wdl]: you shouldn't need this second pipefail command since it is already on line 243

File: annotation_full.wdl [hunk 2]
```diff
@@ -245,6 +264,50 @@ task stage {
...
+    String  prefix=sub(proj_id, ":", "_")
+    File input_file
+    String container
+    String output_file = "${prefix}_map.fasta"
+    Int min_seq_length = 150      # default value
+    Int unknown_gap_length = 100  # default value
+
+
+  command <<<
+  find_prefix=`grep ${proj_id} ${input_file} | head -1`
```
Comment (aclum) [inline annotation_full.wdl:277]: Why can't we just get this info from the input json proj variable?

File: annotation_full.wdl [hunk 3]
```diff
@@ -10,29 +10,39 @@ workflow annotation {
   String?  gm_license="/refdata/licenses/.gmhmmp2_key"
   Int     additional_threads=16
   String  container="microbiomedata/img-omics@sha256:d5f4306bf36a97d55a3710280b940b89d7d4aca76a343e75b0e250734bc82b71"
+  String  finish_ano_container = "microbiomedata/workflowmeta:1.1.1"
```
Comment (aclum) [inline annotation_full.wdl]: Is there a reason we aren't using microbiomedata/img-omics@sha256:d5f4306bf36a97d55a3710280b940b89d7d4aca76a343e75b0e250734bc82b71?

File: annotation_full.wdl [hunk 4]
```diff
@@ -610,7 +689,12 @@ task finish_ano {
 
        ln ${ano_info_file} ${prefix}_imgap.info
```
Comment (aclum) [inline annotation_full.wdl]: need a failover to a soft link

File: annotation_full.wdl [hunk 5]
```diff
@@ -610,7 +689,12 @@ task finish_ano {
...
 
-   }
+       if [[ "${map_execute}" = true ]]
+        then
+        ln ${map_file} ${prefix}_contig_names_mapping.tsv
```
Comment (aclum) [inline annotation_full.wdl]: line 694 and 695 need failovers to softlinks

File: structural-annotation.wdl [hunk 6]
```diff
@@ -21,16 +21,20 @@ workflow s_annotate {
...
+
 
   if(pre_qc_execute) {
     call pre_qc {
       input:
         project_type = imgap_project_type,
         input_fasta = imgap_input_fasta,
         project_id = imgap_project_id,
-        container=container
+        container=pre_qc_container
```
Comment (aclum) [inline structural-annotation.wdl]: same question here about containers

**Review by aclum — COMMENTED** [review]
See individual comments.

File: annotation_full.wdl (hunk 1 above)
Comment (kaijli) [reply annotation_full.wdl]: @scanon added this one, i don't quite remember why, but something about the following line could fail if not?

File: annotation_full.wdl (hunk 2 above)
Comment (kaijli) [reply annotation_full.wdl:277]: checks for whether the prefix is there or not to decrease the number of variables that need human input.

File: annotation_full.wdl (hunk 3 above)
Comment (kaijli) [reply annotation_full.wdl]: this was the workflow container listed below in the finish_ano section, so i brought it to the top. not sure why we are not using the main container

File: annotation_full.wdl (hunk 4 above)
Comment (kaijli) [reply annotation_full.wdl]: missed this one, thanks

File: structural-annotation.wdl (hunk 6 above)
Comment (kaijli) [reply structural-annotation.wdl]: pre-qc container was what was listed at some point, i'll test both instances with the main container and remove if no issues pop up

**Comment by kaijli** [comment]
made requested changes, jaws id 79268

**Comment by aclum** [comment]
Looks good.

**Pull request merged**